gulpfile.js

```diff
@@ -1,110 +1,96 @@
-		.pipe(header('\n/* **********************************************\n' +
-			'     Begin <%= file.relative %>\n' +
-			'********************************************** */\n\n'))
-		.pipe(concat('prism.js'))
-		.pipe(gulp.dest('./'));
+const { src, dest, series, parallel, watch } = require('gulp');
+
+const rename = require('gulp-rename');
+const uglify = require('gulp-uglify');
+const header = require('gulp-header');
+const concat = require('gulp-concat');
+const replace = require('gulp-replace');
+const pump = require('pump');
+const fs = require('fs');
+
+const paths = {
+	componentsFile: 'components.json',
+	componentsFileJS: 'components.js',
+	components: ['components/**/*.js', '!components/index.js', '!components/**/*.min.js'],
+	main: [
+		'components/prism-core.js',
+		'components/prism-markup.js',
+		'components/prism-css.js',
+		'components/prism-clike.js',
+		'components/prism-javascript.js',
+		'plugins/file-highlight/prism-file-highlight.js'
+	],
+	plugins: ['plugins/**/*.js', '!plugins/**/*.min.js'],
+	showLanguagePlugin: 'plugins/show-language/prism-show-language.js',
+	autoloaderPlugin: 'plugins/autoloader/prism-autoloader.js',
+	changelog: 'CHANGELOG.md'
+};
+
+const componentsPromise = new Promise((resolve, reject) => {
+	fs.readFile(paths.componentsFile, {
+		encoding: 'utf-8'
+	}, (err, data) => {
+		if (!err) {
+			resolve(JSON.parse(data));
+		} else {
+			reject(err);
+		}
+	});
 });
 
-function plugins(cb) {
-	pump(
-		[
-			gulp.src(paths.plugins),
-			inlineRegexSource(),
-			uglify(),
-			rename({ suffix: '.min' }),
-			gulp.dest('plugins')
-		],
-		cb
+function inlineRegexSource() {
+	return replace(
+		/\/((?:[^\n\r[\\\/]|\\.|\[(?:[^\n\r\\\]]|\\.)*\])*)\/\.source\b/g,
+		(m, source) => {
+			// escape backslashes
+			source = source.replace(/\\/g, '\\\\');
+			// escape single quotes
+			source = source.replace(/'/g, "\\'");
+			// unescape characters like \\n and \\t to \n and \t
+			source = source.replace(/(^|[^\\])\\\\([nrt0])/g, '$1\\$2');
+			// wrap source in single quotes
+			return "'" + source + "'";
+		}
 	);
 }
 
-gulp.task('components-json', function (cb) {
-	componentsPromise.then(function (data) {
-		data = 'var components = ' + JSON.stringify(data) + ';\n' +
-			'if (typeof module !== \'undefined\' && module.exports) { module.exports = components; }';
-		fs.writeFile(paths.componentsFileJS, data, cb);
+function minifyJS() {
+	return [
+		inlineRegexSource(),
+		uglify()
+	];
+}
+
+
+function minifyComponents(cb) {
+	pump([src(paths.components), ...minifyJS(), rename({ suffix: '.min' }), dest('components')], cb);
+}
+function minifyPlugins(cb) {
+	pump([src(paths.plugins), ...minifyJS(), rename({ suffix: '.min' }), dest('plugins')], cb);
+}
+function build(cb) {
+	pump([src(paths.main), header('\n/* **********************************************\n' +
+		'     Begin <%= file.relative %>\n' +
+		'********************************************** */\n\n'), concat('prism.js'), dest('./')], cb);
```

Comment:
If we're ES6-ifying everything, we can make this a template string.

Reply:
```suggestion
********************************************** */

`), concat('prism.js'), dest('./')], cb);
```

Reply:
I didn't do this because then the line ends of `prism.js` depend on the line ends of `gulpfile.js`.

But if you think that that's not a problem I'll change it.

Reply:
nvm. Template strings ftw! I already use them anyway for `components.js`.